src/xpk/core/cluster_private.py

```diff
@@ -0,0 +1,170 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from .core import zone_to_region
+from .commands import run_command_for_value, run_command_with_updates
+from ..utils.console import xpk_exit, xpk_print
+from ..utils.network import add_current_machine_to_networks, is_current_machine_in_any_network
+
+
+def authorize_private_cluster_access_if_necessary(args) -> int:
+  """Updates a GKE cluster to add authorize networks to access a private cluster's control plane, if not added already.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and error code otherwise.
+  """
+  if not is_cluster_private(args):
+    if not args.private and args.authorized_networks is None:
+      xpk_print('Cluster is public and no need to authorize networks.')
+      return 0
+    else:
+      xpk_print(
+          'Cannot convert an existing public cluster to private. The arguments'
+          ' --private and --authorized-networks are not acceptable for public'
+          ' clusters.'
+      )
+      return 1
+
+  new_authorized_networks_needed = args.authorized_networks is not None
+
+  authorized_networks = (
+      args.authorized_networks
+      if new_authorized_networks_needed
+      else get_cluster_authorized_networks(args)
+  )
```

Comment:
Should we maybe add an additional check comparing the existing authorized networks with the ones provided by the user? It seems that this code will change the authorized networks even if the provided ones are the exact same as the ones currently existing.

Reply:
If users set this, it wouldn't be problem to update it.